cloud/google/machineactuator.go

```diff
@@ -112,27 +111,15 @@ func NewMachineActuator(kubeadmToken string, machineClient client.MachineInterfa
 		return nil, err
 	}
 
-	// Only applicable if it's running inside machine controller pod.
-	var privateKeyPath, user string
-	if _, err := os.Stat("/etc/sshkeys/private"); err == nil {
-		privateKeyPath = "/etc/sshkeys/private"
-
-		b, err := ioutil.ReadFile("/etc/sshkeys/user")
-		if err == nil {
-			user = string(b)
-		} else {
-			return nil, err
-		}
-	}
-
 	return &GCEClient{
 		computeService: computeService,
 		scheme:         scheme,
 		codecFactory:   codecFactory,
 		kubeadmToken:   kubeadmToken,
 		sshCreds: SshCreds{
-			privateKeyPath: privateKeyPath,
-			user:           user,
+			privateKeyPath: sshPrivateKeyPath,
```

Comment:
It seems like one can supply a private key with an invalid/missing public key and vice versa. I suspect the behavior will be cryptic errors related to "unable to connect" or require log diving. It would be preferable to give users early feedback that they have done so, i.e. validate when the application is started or something along those lines.